airgun/entities/ansible_role.py

```diff
@@ -0,0 +1,57 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.views.ansible_role import AnsibleRolesImportView
+from airgun.views.ansible_role import AnsibleRolesView
+
+
+class AnsibleRolesEntity(BaseEntity):
+    endpoint_path = '/ansible/ansible_roles'
+
+    def search(self, value):
+        """Search for existing Ansible Role"""
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def delete(self, entity_name, values):
+        """Delete Ansible Role from Satellite"""
+        view = self.navigate_to(self, 'All')
+        view.search(entity_name)
+        view.table.row(name=entity_name)['Actions'].widget.fill('Delete')
+        self.browser.handle_alert()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def count_imported_roles(self):
+        view = self.navigate_to(self, 'All')
+        return view.total_imported_roles.read()
+
+    def import_all_roles(self):
+        view = self.navigate_to(self, 'Import')
+        available_roles_count = view.total_available_roles.read()
+        view.select_all.fill(True)
+        view.submit.click()
+        return available_roles_count
+
+
+@navigator.register(AnsibleRolesEntity, 'All')
+class ShowAllRoles(NavigateStep):
+
+    VIEW = AnsibleRolesView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Configure', 'Roles')
+
+
+@navigator.register(AnsibleRolesEntity, 'Import')
+class ImportAnsibleRole(NavigateStep):
+    """Navigate to the Import Roles page"""
+
+    endpoint_path = '/ansible/ansible_roles'
+    VIEW = AnsibleRolesImportView
+
+    def prerequisite(self, *args, **kwargs):
```

Comment:
You can use `NavigateToSibling` here to flatten this quite a bit. I don't think its in wide use in airgun, you followed the 'regular' pattern here.

```
from navmazing import NavigateToSibling # at the top of the file

# then within the class, where you currently are defining prerequisite as a function

prerequisite = NavigateToSibling('All')
```

The class takes care of resolving the object, and navigating at the right time. 

https://github.com/RedHatQE/navmazing/blob/master/navmazing/__init__.py#L165